### DISCUSSION TITLE
CheckItem placeholders

### DISCUSSION THREAD

**Discussion by Tanguygab** [Zrips/CMI#8083] [Enhancement]
Since CheckItem 2.7.1, the give and remove placeholders were blocked behind a config setting which is set to false by default.
Though because of CMI's hardblock of the placeholders, it's not possible to use them at all.
Not only does it prevent checkitem placeholders to work (even the normal ones that don't give nor remove items), it also breaks other placeholders, on the same line, though I'm guessing it's intended since you'd expect to remove the checkitem placeholder and it'll work, but it would be nice to check only if the placeholder startswith `%checkitem_remove` or `%checkitem_give`. 
![image](https://github.com/Zrips/CMI/assets/45695833/d5f7a0a2-ed7b-450b-9dbc-dcc5979156dc)

All in all, considering CheckItem added an option have those disabled by default, could the hardblock be removed? Or if you worry that people won't update their expansion, at least add a config option to enable the placeholders.

**Comment by ZelefRU** [comment]
Yes, I ran into the same problem, I didn't even know about the existence of "give" and "remove", and used only checking items. It's disappointing that now some of my mechanics are not working because of the blocking of this Placeholder :'(

**Comment by 1470429149** [comment]
I noticed that %checkitem% is banned by CMI due to security issues
However, the latest version of %checkitem% adds switches for the "give" and "remove" functions in the "placeholderapi" configuration file.
We should turn off %checkitem% by default in CMI
You should also add a function to enable %checkitem% in the configuration file, and tell CMI users that after enabling "%checkitem%", players can also use the security issue of %checkitem% placeholders.

checkitem can help users greatly enhance the scalability of CMI. I hope this suggestion can be accepted.

**Reply by Tanguygab** [reply]
Yes, that's what I said... ![image](https://github.com/Zrips/CMI/assets/45695833/93e7f810-3dd9-40b9-b493-06f663cae0ef)
